lib/wallet/walletdb.js

```diff
@@ -206,6 +206,65 @@ class WalletDB extends EventEmitter {
       wallet.id, wallet.wid, addr.toString(this.network));
 
     this.primary = wallet;
+
+    await this.migrateChange();
+  }
+
+  /**
+   * Run change address migration.
+   * @param {Wallet} wallet
+   */
+
+  async migrateChange() {
+    if (this.options.memory)
+      return;
+
+    if (await this.db.has(layout.M.encode(0)))
+      return;
+
+    const b = this.db.batch();
+
+    b.put(layout.M.encode(0), null);
+
+    this.logger.warning('Checking change address corruption...');
+
+    const wids = await this.db.keys({
+      gte: layout.W.min(),
+      lte: layout.W.max(),
+      parse: key => layout.W.decode(key)[0]
+    });
+
+    let total = 0;
+
+    for (const wid of wids) {
+      const wallet = await this.get(wid);
+
+      this.logger.info('Checking wallet (id=%s, wid=%d).',
+                       wallet.id, wid);
+
+      total += await wallet.migrateChange(b);
+    }
+
+    if (this.options.migrate === 0 || total === 0)
+      await b.write();
```

Comment:
Where does `b` get written to disk in the other case?

Reply:
It doesn't. It only gets written if either 

1) nothing is wrong, just save the new migration ID
or
2) the flag is set -- at this point the batch contains all the new change addresses. but we don't write them unless the user says to

Reply:
The migration index set to 0 needs to be written to disk either way after this process so that migrating from 0 to 1 is a more simple process and so that this logic is cut short on additional restarts.

Reply:
It should only get written when the wallet corruption is fixed, or if there is nothing to fix. hsd will throw an error if the user refuses to repair their DB, so I don't know how they will even get from 0 to 1 before getting to 0 first.